packages/ng-primitives/schematics/checkbox/files/checkbox.component.ts.template

```diff
@@ -0,0 +1,20 @@
+import { Component } from '@angular/core';
+import { NgpCheckbox } from 'ng-primitives/checkbox';
+
+
+@Component({
```

Comment:
So now that you have moved on form controls this is a good opportunity for us to provide some of the additional boilerplate surrounding building custom form controls, for example, integration with Angular Forms.

In the case of a checkbox its important for the `checked` state to be set and updated, so we can also provide a `model` for allowing the value to be set and hook it up to the underlying host directive, e.g.:

```ts
import { model, Component, signal } from '@angular/core';
import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
import { NgpCheckbox } from 'ng-primitives/checkbox';
import { onChange } from 'ng-primitives/utils';

@Component({
  standalone: true,
  selector: 'app-checkbox',
  hostDirectives: [
    {
      directive: NgpCheckbox,
      inputs: ['ngpCheckboxChecked:checked'],
      outputs: ['ngpCheckboxCheckedChange:checkedChange'],
    },
  ],
  providers: [
    { provide: NG_VALUE_ACCESSOR, useExisting: CheckboxComponent, multi: true },
  ],
  styles: `
    :host {
    }
    
    :host[data-hover='true'] {
    }
    
    :host[data-checked='true'] {
    }

    :host[data-focus-visible='true'] {
    }
  `,
  template: `
    @if (checked()) {
      <!-- Add checkmark here -->
    }
  `,
  host: {
    '(focusout)': 'onTouched?.()',
  },
})
export class CheckboxComponent implements ControlValueAccessor {
  /**
   * The checked state of the checkbox.
   */
  readonly checked = model<boolean>(false);

  /**
   * The onChange callback.
   */
  private onChange?: (value: boolean) => void;

  /**
   * The onTouched callback.
   */
  protected onTouched?: () => void;

  constructor() {
    onChange(this.checked, (value) => {
      if (value !== undefined) {
        this.onChange?.(value!);
      }
    });
  }

  writeValue(checked: boolean): void {
    this.checked.set(checked);
  }

  registerOnChange(fn: (value: boolean) => void): void {
    this.onChange = fn;
  }

  registerOnTouched(fn: () => void): void {
    this.onTouched = fn;
  }
}
```

Here is a stackblitz for reference: https://stackblitz.com/edit/angular-es2sm2?file=src/main.ts

Any questions or issues let me know!

Reply:
Done. 
I just used the `forwardRef ` to avoid the error of "used before its declaration"